vms/platformvm/txs/builder/backend.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package builder
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/utils/set"
+	"github.com/ava-labs/avalanchego/vms/components/avax"
+	"github.com/ava-labs/avalanchego/vms/platformvm/fx"
+	"github.com/ava-labs/avalanchego/vms/platformvm/state"
+	"github.com/ava-labs/avalanchego/wallet/chain/p/signer"
+
+	walletbuilder "github.com/ava-labs/avalanchego/wallet/chain/p/builder"
+)
+
+var (
+	_ walletbuilder.Backend = (*Backend)(nil)
+	_ signer.Backend        = (*Backend)(nil)
+)
+
+func NewBackend(
+	addrs set.Set[ids.ShortID],
+	state state.State,
+	atomicUTXOsMan avax.AtomicUTXOManager,
+) *Backend {
+	return &Backend{
+		addrs:          addrs,
+		state:          state,
+		atomicUTXOsMan: atomicUTXOsMan,
+	}
+}
+
+type Backend struct {
+	addrs          set.Set[ids.ShortID]
+	state          state.State
+	atomicUTXOsMan avax.AtomicUTXOManager
+}
+
+func (b *Backend) UTXOs(_ context.Context, sourceChainID ids.ID) ([]*avax.UTXO, error) {
+	if sourceChainID == constants.PlatformChainID {
+		return avax.GetAllUTXOs(b.state, b.addrs)
+	}
+
+	atomicUTXOs, _, _, err := b.atomicUTXOsMan.GetAtomicUTXOs(sourceChainID, b.addrs, ids.ShortEmpty, ids.Empty, MaxPageSize)
```

Comment:
It seems like we should be fetching all UTXOs here - not just the first `MaxPageSize` right?